File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> arr) {
    int answer = 1, flag = 0;
    
    for(int i=0; i<arr.size(); i++) {
        for(int j=0; j<arr.size(); j++){
            if(arr[i][j] != arr[j][i]){
                answer = 0;
                flag = 1;
                break;
            }
        }
        if(flag == 1){
            break;
        }
    }
    
    return answer;
}

/*
    배열의 길이 = 배열의 원소의 길이
    
*/

File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.swift
import Foundation

func solution(_ arr:[[Int]]) -> Int {
    
    for i in 0..<arr.count {
        for j in 0..<arr.count {
            if arr[i][j] != arr[j][i] {
                return 0
            }
            
        }
    }
    
    return 1
}